src/pages/EditRequestDescriptionPage.js

```diff
@@ -77,7 +77,7 @@ function EditRequestDescriptionPage({defaultDescription, onSubmit}) {
                         }}
                         autoGrowHeight
                         containerStyles={[styles.autoGrowHeightMultilineInput]}
-                        submitOnEnter={!Browser.isMobile()}
+                        submitOnEnter={!canUseTouchScreen()}
```

Comment:
I don't think that `submitOnEnter` is a good name, especially in the context of our recent changes

Reply:
We should also add a comment that this is a property intended only for effectively multiline input

Reply:
Can you suggest a new name which you seem fit here?
And I will add a comment indicating the same.

Reply:
I think we can "hold" this thread on [this topic](https://github.com/Expensify/App/pull/34471#discussion_r1461807307), as we might just nuke this property.

Reply:
Not really. We are not allowing submit on enter on every multiline input even on keyboard devices (desktop). So we need to decide a name here.

I am thinking `forceSubmitOnEnter` would be better.

Reply:
I'm having a problem with the "Enter" part. I know that it's a mess.

In my mental model, the submitting key ("Go" / "Done" / (arrow) ...) is presented _instead_ of the Enter/Return key).

If I recall correctly, pressing the submitting key on the virtual keyboard has a different effect from pressing the actual "Enter" key on the physical external Bluetooth keyboard connected to an iPhone or iPad. Possibly similar on Android devices.

This is complicated by the mobile Web browsers, which don't expose the native submit behavior to DOM API. This might be a backwards-compatibility leftover from times when all touch events could only be handled by DOM mouse events.

That's why I'm insisting on dropping the "Enter" part from the name and focusing on the model where a text input can be "submitted". On the Web (`react-native-web`) this is implemented as listening for the "Enter" key (which on native iOS/Android, again, _could_ be the press of the virtual submitting key) from a lack of a better option.

Reply:
Ok should we call it `forceSubmitOnReturnKey` as RN also calls it a Return Key?

Reply:
This logic has a flaw. It will be always be false for multi-line inputs.

Reply:
Not what I meant, and I don't think the latest change is reasonable 🙁

I meant this to be the _centralized_ logic we discussed before:

```
inputAllowsSubmit = single-line -> true; multi-line -> is_desktop_web
```

Just a local, in-component conditional computation. 

And _this_, as I stated explicitly, I meant to be the component property. This is the reincarnation of `submitOnEnter`, with an adjusted contract:

```
property shouldSubmitForm (default: for single-line -> true; for multi-line -> false)
```

Technically, it won't be a "React default", as it would be tricky technically. Technically, it will be `undefined` by default in the React terms, and then we'll calculate an effective default inside the component.

Try implementing this, or raise concerns if you disagree with it. If you are still unsure what I mean, I can implement the commit and let you cherry-pick it.

Reply:
@shubham1206agra Please let me know if I worded this clearly

Reply:
I may be misunderstanding something here. Can you give me commit diff, and then we can discuss this?

Reply:
Would you review it and test it? I could've missed something.

Reply:
Your logic is mostly fine with some minor bugs, which I am fixing.